src/scrabble.js

```diff
@@ -1,5 +1,231 @@
-function scrabble() {
+const letterValues = {
+  a: 1,
+  e: 1,
+  i: 1,
+  o: 1,
+  u: 1,
+  l: 1,
+  n: 1,
+  r: 1,
+  s: 1,
+  t: 1,
+  d: 2,
+  g: 2,
+  b: 3,
+  c: 3,
+  m: 3,
+  p: 3,
+  f: 4,
+  h: 4,
+  v: 4,
+  w: 4,
+  y: 4,
+  k: 5,
+  j: 8,
+  x: 8,
+  q: 10,
+  z: 10
+}
+
+//  Calculates the base score of a string with no conditions applied
+//  If checked character does not have a number value of 1 or greater, it doesn't exist in the letterValues array and is skipped
+
+function baseScore(givenWord) {
+  const lowerCaseString = givenWord.toLowerCase()
+  let wordScore = 0
+  for (let i = 0; i < lowerCaseString.length; i++) {
+    const letterScore = letterValues[lowerCaseString[i]]
+    if (letterScore >= 1) {
+      wordScore += letterScore
+    }
+  }
+  return wordScore
+}
+
+//  Check if double or triple word
+//  Compares first and last characters to look for a set of brackets
+
+function isDoubleWord(givenWord) {
+  if (givenWord[0] === `{` && givenWord[givenWord.length - 1] === `}`) {
+    return true
+  } else {
+    return false
+  }
+}
+
+function isTripleWord(givenWord) {
+  if (givenWord[0] === `[` && givenWord[givenWord.length - 1] === `]`) {
+    return true
+  } else {
+    return false
+  }
+}
+
+//  Checks for two sets of brackets
+
+function isDoubleAndTripleWord(givenWord) {
+  if (
+    (givenWord[0] === `{` &&
+      givenWord[givenWord.length - 1] === `}` &&
+      givenWord[1] === `[` &&
+      givenWord[givenWord.length - 2] === `]`) ||
+    (givenWord[1] === `{` &&
+      givenWord[givenWord.length - 2] === `}` &&
+      givenWord[0] === `[` &&
+      givenWord[givenWord.length - 1] === `]`)
+  ) {
+    return true
+  } else {
+    return false
+  }
+}
+
+//  Solves the edge case {d}o{g}
+//  Checks if the 1st & 3rd characters are a bracket and if the last and last - 2 characters are a bracket
+//  Horribliy clunky but it works!
+
+function endLetterMultipliers(givenWord) {
+  if (
+    (givenWord[0] === `{` &&
+      givenWord[2] === `}` &&
+      givenWord[givenWord.length - 3] === `{` &&
+      givenWord[givenWord.length - 1] === `}`) ||
+    (givenWord[0] === `[` &&
+      givenWord[2] === `]` &&
+      givenWord[givenWord.length - 3] === `[` &&
+      givenWord[givenWord.length - 1] === `]`) ||
+    (givenWord[0] === `[` &&
+      givenWord[2] === `]` &&
+      givenWord[givenWord.length - 3] === `{` &&
+      givenWord[givenWord.length - 1] === `}`) ||
+    (givenWord[0] === `{` &&
+      givenWord[2] === `}` &&
+      givenWord[givenWord.length - 3] === `[` &&
+      givenWord[givenWord.length - 1] === `]`)
+  ) {
```

Comment:
Yeah, I realised after I finished it that I could have used regex to do this but at the time I just went with an if statement instead! As you said, if it works it works, even if it could have been a lot cleaner!